plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/AbstractTestOracleTypeMapping.java

```diff
@@ -727,12 +729,182 @@ public void testTimestamp(ZoneId sessionZone)
+                .addRoundTrip("TIMESTAMP(3)", "TIMESTAMP '2020-09-27 12:34:56.999'", createTimestampType(3), "TIMESTAMP '2020-09-27 12:34:56.999'")
+                .addRoundTrip("TIMESTAMP(7)", "TIMESTAMP '2020-09-27 12:34:56.1234567'", createTimestampType(7), "TIMESTAMP '2020-09-27 12:34:56.1234567'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.123456789'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123456789'")
+
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.000000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.1'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.100000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.9'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.900000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.123'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.123000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.123000'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.123000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.999'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.999000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.1234567'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.123456700'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '1970-01-01 00:00:00.123456789'", createTimestampType(9), "TIMESTAMP '1970-01-01 00:00:00.123456789'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.1'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.100000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.9'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.900000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.123'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.123000'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.999'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.999000000'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.1234567'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123456700'")
+                .addRoundTrip("TIMESTAMP(9)", "TIMESTAMP '2020-09-27 12:34:56.123456789'", createTimestampType(9), "TIMESTAMP '2020-09-27 12:34:56.123456789'")
+
+                .execute(getQueryRunner(), oracleCreateAndInsert("test_ts_oracle"));
```

Comment:
Can we have additional test coverage for `CREATE` table with a higher precision and check on `SHOW CREATE TABLE`

Reply:
test added